### PR TITLE
Added video player to each incident page, made changes to reports and submissions schema and added media upload to the submission page.

### PR DESCRIPTION
**Pull request by virasakd**
1) In an abundance of files, we renamed image_url to media_url to match our schema changes for the report and submission collections. 
2) In the submission and reports schema, image_url has been renamed to media_url. 
3) Changed the image upload component on the submission page to media upload. Users can now submit videos by providing a URL link to a youtube or Vimeo video. These modifications were made to submitform.js, PreviewImageInput.js, and cloudinary.js. 
4) The file check_video.js was added to ensure that the video submitted by the user is accessible. 
5) In SubmissionReview.js, changes were made to fix the invalid null reads by submission.incident_ids.length === 0; 
6) Image_url in Graphql queries and gatsby-config.js, was changed to media_url.
7) Added a Video Player component that adds an iFrame to play YouTube and Vimeo videos, powered by React Player (which uses iFrame APIs of several popular video sites). If the media_url provided is NOT a video, can't be played, or has been made private, then falls back.
 8) Added this Video Player to Incident Report Cards (ReportCard) and the Details view of the Discover page, with the video thumbnails as a fallback. 
9) Added a Video Player Card to the Incident page. If an incident has a video URL, it will appear, showing the video from the incident's earliest report that includes a video. If this video can't be played, it uses a recursive component to show the next video. 
10) Currently missing: Cypress.js tests.

### CODE REVIEW

**Comment by smcgregor** [comment]
Thanks @virasakd, can you indicate the scope of the requested review? Something like:

1. What the functionality should be doing in user space, and why
2. Whether there is any work to do before this should be mergeable beyond adding test coverage
3. How to test this as a user in a deploy preview

@pdcp1 once someone does the above, can you take first code/functional review? There is not much time left in the project so we need to turn the review around quickly. In scope are changes for stability, style, and consistency. Feature-driven changes are in scope for the review, but unless they are critical they will be put into an issue for future work.

File: site/gatsby-site/src/components/discover/hitTypes/Details.js
Comment (pdcp1) [inline site/gatsby-site/src/components/discover/hitTypes/Details.js:53]: Is `itemIdentifier` field needed? It looks like `<Image>` component doesn't have a field with that name:
image

File: site/gatsby-site/src/components/forms/PreviewImageInputGroup.js
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/PreviewImageInputGroup.js:42]: This commented line should be removed

File: site/gatsby-site/src/components/forms/PreviewImageInputGroup.js
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/PreviewImageInputGroup.js:37]: This function is duplicated on `/src/components/cite/VideoPlayer.js`. Please refactor this to use just one function.

File: site/gatsby-site/src/components/forms/check_video.js
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/check_video.js:1]: This line should be removed

File: site/gatsby-site/src/components/forms/check_video.js
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/check_video.js:2]: This file should be moved to the `/src/utils` folder

File: site/gatsby-site/src/components/forms/SubmitForm.js
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/SubmitForm.js:186]: This comment looks outdated

File: site/gatsby-site/src/components/forms/SubmitForm.js
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/SubmitForm.js:182]: Change this to `if (!isValid) {`

**Review by pdcp1 — CHANGES_REQUESTED** [review]

File: site/gatsby-site/src/components/cite/VideoPlayer.js
Comment (ShaurGaur) [reply site/gatsby-site/src/components/cite/VideoPlayer.js:17]: Deleting the `twitterRegex` line now.

File: site/gatsby-site/src/components/discover/hitTypes/Details.js
Comment (ShaurGaur) [reply site/gatsby-site/src/components/discover/hitTypes/Details.js:53]: From a search of the codebase, all other instances of `<Image>` also have `itemIdentifier`, so we kept it for consistency.

**Comment by pdcp1** [comment]
Please let me know when all the comments above are done in order to do another code review round. Code conflicts with Staging should be resolved as well.

**Comment by virasakd** [comment]
@pdcp1 All comments and code conflicts, besides package-lock.json, have been resolved.

**Comment by pdcp1** [comment]
> @pdcp1 All comments and code conflicts, besides package-lock.json, have been resolved.

@virasakd The `package-lock.json` should be merged as well. To do that, please do an `npm install` on your side and commit/push the `package-lock.json` generated.

File: site/gatsby-site/migrations/2022.06.07T19.46.03.cloudinary-id-update.js
Comment (smcgregor) [inline site/gatsby-site/migrations/2022.06.07T19.46.03.cloudinary-id-update.js:21]: Two issues here. 1. Don't change historical migrations. 2. This orphans all the data stored at `image_url`. You need to migrate the values by setting the new value and unsetting the old value. This won't be mergeable without that.

**Comment by smcgregor** [comment]
I left one comment inline that is important. We should also proceed with a functional review on a deploy preview. Do you have a version of the changes hosted somewhere?

**Comment by pdcp1** [comment]
If you need any help or have any questions on creating a new migration, please reach me on Slack and I can assist you with some guidance. I recommend you read the README file first.

File: site/gatsby-site/migrations/2022.06.07T19.46.03.cloudinary-id-update.js
Comment (smcgregor) [reply site/gatsby-site/migrations/2022.06.07T19.46.03.cloudinary-id-update.js:21]: Yes. When you change a historically applied migration, you are breaking the history of changes and the databases will be in a more uncertain state.

**Comment by NotNihalx** [comment]
> Thanks @virasakd, can you indicate the scope of the requested review? Something like:
> 
> 1. What the functionality should be doing in user space, and why
> 2. Whether there is any work to do before this should be mergeable beyond adding test coverage
> 3. How to test this as a user in a deploy preview
> 
> @pdcp1 once someone does the above, can you take first code/functional review? There is not much time left in the project so we need to turn the review around quickly. In scope are changes for stability, style, and consistency. Feature-driven changes are in scope for the review, but unless they are critical they will be put into an issue for future work.

1. The primary functionality in the user space is to be able to add media support and see said media after the submission is complete.

2. As of now, outside of adding unit tests, there is no major work aside from adding tests and addressing Merge comments.

3. As a user, this can be tested by submitting an incident with media links (or lack thereof) of your choice and then going to the view submission page to see if the content is properly rendered and is playable or not. Additionally, more testing can be done by deleting or setting a video to private after a submission upload and then making sure that this video is later correctly  non-accessible with a reason in the view incidents page.

File: site/gatsby-site/package.json
Comment (pdcp1) [inline site/gatsby-site/package.json:26]: I don't see any reference to this package in the code. If this is not used please remove and re-generate the `package-lock.json` file.

File: site/gatsby-site/package.json
Comment (pdcp1) [inline site/gatsby-site/package.json:61]: I don't see any reference to this package in the code. If this is not used please remove and re-generate the `package-lock.json` file.

**Review by pdcp1 — CHANGES_REQUESTED** [review]

File: site/gatsby-site/package.json
Comment (virasakd) [reply site/gatsby-site/package.json]: removed babel-plugin-lodash

File: site/gatsby-site/package.json
Comment (virasakd) [reply site/gatsby-site/package.json]: removed lmdb

File: site/gatsby-site/package.json
Comment (pdcp1) [inline site/gatsby-site/package.json:110]: We should keep the greatest version `^3.6.2` of `@netlify/plugin-gatsby` package

**Review by pdcp1 — CHANGES_REQUESTED** [review]

File: site/gatsby-site/package.json
Comment (virasakd) [reply site/gatsby-site/package.json]: Alright, I changed @netlify/plugin-gatsby back to ^3.6.2.

**Review by pdcp1 — CHANGES_REQUESTED** [review]

**Comment by ShaurGaur** [comment]
@pdcp1 Pending Cypress test updates and resolving new merge conflicts (currently in progress), **we are ready for a final review**. 

**Main front-end changes:**

- Step 2 of the Submit Form allows video URL submissions (in "Media Address") from YouTube and Vimeo, and shows video thumbnails which are saved to Cloudinary.
- Video player for reports (when included) in the Details view of Discover page and ReportCards in the Cite page. *The video overflows the report card box when resized, but Sean has indicated that this can be dealt with as an issue later.*
- For each incident with a video, the video from its oldest report in an "Incident Video" player card at the top of the Cite page.

To view these changes displayed visually, look at this Netlify deploy: https://codefreeze--dfid.netlify.app/, based off of a branch derived from `MergeStagingUpload`. 

- Its Landing page doesn't load due to a problem with our workaround for only including 39 incidents in our scraped database. But this isn't in `MergeStagingUpload` so shouldn't affect our PR. We don't change Landing or WordCounts otherwise.

**Overview of main file changes:**

- Many files just change `image_url` to `media_url`.
- Video Player component: `site/gatsby-site/src/components/cite/VideoPlayer.js`
    - Used in ReportCard: `site/gatsby-site/src/components/reports/ReportCard.js`
    - Used in Details in Discover: `site/gatsby-site/src/components/discover/hitTypes/Details.js`
- Incident Video Card: `site/gatsby-site/src/components/cite/VideoPlayerCard.js`
    - Used in cite page: `site/gatsby-site/src/templates/citeTemplate.js`
- `utils/cloudinary.js` for thumbnail extraction.
- `utils/video.js` for video check utilities.

Let us know if you have any questions or comments!

**Comment by pdcp1** [comment]
@ShaurGaur Could you please provide auth credentials for a user with at least `incident_editor` role on your site?

**Comment by ShaurGaur** [comment]
> 

@pdcp1 For [https://codefreeze--dfid.netlify.app/](https://codefreeze--dfid.netlify.app/) here's the credentials for a user with `incident_editor` and `admin` roles.

**User email:** gaurs@oregonstate.edu
**Password**: hnq8YVK4fdv.hvk6xjh

**Comment by smcgregor** [comment]
@pdcp1 ^

@ShaurGaur it looks like there are two merge conflicts. I think this is close enough to landing to warrant resolving those.

**Comment by pdcp1** [comment]
@ShaurGaur I’m getting this error when I try to submit a new report on your site https://codefreeze--dfid.netlify.app/
```
{"error":"cannot find app using Client App ID 'aiidstitch2-rxvyw'"}
```
Looking at the browser log console, all the calls to the graphql endpoint are failing with the same error.

**Comment by ShaurGaur** [comment]
@pdcp1 added Cypress tests after checking if they work in our own DB with video URLs. good for a final review!

**Comment by smcgregor** [comment]
@pdcp1 Please let me know if you would like to discuss or move this into staging.

**Comment by pdcp1** [comment]
@ShaurGaur  I see less than 50 incidents on your site. Did you dump and restore the Production DB and run the migration that you created?
We need this in order to prove that the tests passed for a more realistic data.

**Comment by ShaurGaur** [comment]
@pdcp1 yes. we did dump and restore the AIID Production DB internally and ran tests against that. The codefreeze site is just the deepfake incidents we scraped

**Comment by pdcp1** [comment]
@smcgregor We should Accept the last deploy from `MergeStagingUpload` here https://app.netlify.com/sites/staging-aiid/deploys before merge this into staging to see if we can generate a preview link pointing to our staging DB

**Comment by smcgregor** [comment]
Running: https://app.netlify.com/sites/staging-aiid/deploys/648117f7d9e1230008648dd3

**Comment by smcgregor** [comment]
@pdcp1 failed due to the schema. Did the migration not run? https://app.netlify.com/sites/staging-aiid/deploys/648117f7d9e1230008648dd3

**Comment by pdcp1** [comment]
@smcgregor Actually, it's OK if it fails on Staging since the PR includes schema changes. We can't use our preview strategy to do the functional review, we have to use their environment. I did a functional review on their site https://codefreeze--dfid.netlify.app/ but maybe it's better to have another pair of eyes before merging it to staging.
Maybe @lmcnulty @clari182 @cesarvarela

**Comment by smcgregor** [comment]
@cesarvarela can you do a final review here and merge if you are 👍 ?

**Comment by cesarvarela** [comment]
closing in favor of #2143

**Pull request closed without merging**
